sortVariants.worker.js

```diff
@@ -1,18 +1,32 @@
-import { scoreResult } from './app/decode/common'
+import { scoreResult, quickSortByScore } from './app/decode/common'
 
 onmessage = function (e) {
   if (!e) return
-  const { variants, firstItem, pageSize } = e.data
-  const lastItem = firstItem + pageSize || variantArray.length - 1
+  const { variants } = e.data
 
-  const results = variants
-    .slice(0, 1)
-    .concat(
-      variants
-        .slice(1)
-        .sort((a, b) => scoreResult(b.decoded) - scoreResult(a.decoded))
+  const baseVariant = variants.slice(0, 1)
+  const variantsToSort = variants.slice(1)
+  const sizes = [500, 10000, variantsToSort.length].filter(
+    (size) => size <= variantsToSort.length
+  )
+  sizes.forEach((size) => {
```

Comment:
Přijde mi zbytečné to při každé iteraci řadit znovu. Kdyby se iterovalo třeba po 500 záznamech až do posledního (1-500, 501-1000, 1001-1500, ...) a ty nové záznamy se přidávaly do již setříděného pole, tak si myslím, že nějaký znatelný čas ušetříme.

Reply:
Asi by nebylo by ani potřeba implementovat quicksort, stačilo by jen vkládat do setříděného pole: https://stackoverflow.com/a/21822316/5259610

Reply:
Je to velmi podobně rychlé. Výhoda je spíš v tom, že je to méně a lépe čitelného kódu, tak jsem to změnil.